src/lib/generate-starterpack.ts

```diff
@@ -0,0 +1,102 @@
+import {AppBskyActorDefs, BskyAgent, Facet} from '@atproto/api'
+import {msg} from '@lingui/macro'
+
+import {logger} from '#/logger'
+import {sanitizeHandle} from 'lib/strings/handles'
+
+export const createStarterPackList = async ({
+  name,
+  description,
+  descriptionFacets,
+  profiles,
+  agent,
+}: {
+  name: string
+  description?: string
+  descriptionFacets?: Facet[]
+  profiles: AppBskyActorDefs.ProfileViewBasic[]
+  agent: BskyAgent
+}): Promise<{uri: string; cid: string}> => {
+  if (profiles.length === 0) throw new Error('No profiles given')
+
+  const list = await agent.app.bsky.graph.list.create(
+    {repo: agent.session!.did},
+    {
+      name,
+      description,
+      descriptionFacets,
+      avatar: undefined,
+      createdAt: new Date().toISOString(),
+      purpose: 'app.bsky.graph.defs#referencelist',
+    },
+  )
+  if (!list) throw new Error('List creation failed')
+  await agent.com.atproto.repo.applyWrites({
+    repo: agent.session!.did,
+    writes: profiles.map(p => ({
+      $type: 'com.atproto.repo.applyWrites#create',
+      collection: 'app.bsky.graph.listitem',
+      value: {
+        $type: 'app.bsky.graph.listitem',
+        subject: p.did,
+        list: list?.uri,
+        createdAt: new Date().toISOString(),
+      },
+    })),
+  })
+
+  return list
+}
+
+export async function generateStarterpack({
```

Comment:
The only reason I didn't do that was because it felt a little weird to need a hook  for this in particular. Not opposed to putting it in a mutation though.